sysfs/system_cpu.go

```diff
@@ -237,3 +240,47 @@ func parseCpufreqCpuinfo(cpuPath string) (*SystemCPUCpufreqStats, error) {
 		SetSpeed:                 stringOut[4],
 	}, nil
 }
+
+func (fs FS) IsolatedCPUs() ([]uint16, error) {
+	isolcpus, err := ioutil.ReadFile(fs.sys.Path("devices/system/cpu/isolated"))
+	if err != nil {
+		return nil, fmt.Errorf("failed to read isolcpus from sysfs: %w", err)
+	}
+
+	return parseIsolCpus(isolcpus)
+}
+
+func parseIsolCpus(data []byte) ([]uint16, error) {
+	isolcpus_str := strings.TrimRight(string(data), "\n")
+
+	var isolcpus_int = []uint16{}
+
+	for _, cpu := range strings.Split(isolcpus_str, ",") {
+		if cpu == "" {
+			continue
+		}
+		if strings.Contains(cpu, "-") {
+			ranges := strings.Split(cpu, "-")
```

Comment:
Check if len(ranges) == 2, otherwise accessing ranges[1] might panic.

Reply:
i thought it'd panic as well, but the split still has 2 items in the slice (`'1-' -> ["1", ""]`)